util/src/main/java/io/kubernetes/client/apimachinery/GroupVersion.java

```diff
@@ -12,24 +12,24 @@
 */
 package io.kubernetes.client.apimachinery;
 
+import static io.kubernetes.client.util.Preconditions.precondition;
+
 import io.kubernetes.client.common.KubernetesObject;
 import io.kubernetes.client.util.Strings;
 import java.util.Objects;
 
 public class GroupVersion {
 
+  private static final GroupVersion GROUP_VERSION_V1 = new GroupVersion("", "v1");
+
   public static GroupVersion parse(String apiVersion) {
-    if (Strings.isNullOrEmpty(apiVersion)) {
-      throw new IllegalArgumentException("No apiVersion found on object");
-    }
     if ("v1".equals(apiVersion)) {
```

Comment:
6. invert the checks: first do the `v1.equals(apiVersion)`, _then_ the `precondition` check. `v1.equals...` check is cheaper than the `precondition` one and I also guess far more often used. As such, it makes a lot more sense to invert these.

Reply:
I have also slightly edited the error messages, so that they make slightly more sense.